Gordon360/Controllers/RecIMControllers/LeagueController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Controllers.RecIMControllers
+{
+    [Route("api/recim")]
+    public class LeagueController : GordonControllerBase
+    {
+        private readonly ILeagueService _leagueService;
+
+        public LeagueController(ILeagueService leagueService)
+        {
+            _leagueService = leagueService;
+        }
+
+        ///<summary>Gets a list of all Leagues</summary>
+        /// <returns>
+        /// All Existing Leagues 
+        /// </returns>
+        [HttpGet]
+        [Route("league")]
+        public ActionResult<IEnumerable<League>> GetAllLeagues()
+        {
+            var result = _leagueService.GetAllLeagues();
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
+
+        ///<summary>Gets a League object by ID number</summary>
+        /// <param name="leagueID">League ID Number</param>
+        /// <returns>
+        /// League object
+        /// </returns>
+        [HttpGet]
+        [Route("league/{leagueID}")]
+        public ActionResult<League> GetLeagueByID(int leagueID)
+        {
+            var result = _leagueService.GetLeagueByID(leagueID);
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
```

Comment:
Similarly, you don't need a NotFound here, you can just return the result (and update the return type to indicate that it can return null.

```suggestion
        public ActionResult<League?> GetLeagueByID(int leagueID)
        {
            var league = _leagueService.GetLeagueByID(leagueID);

            return Ok(league);
        }
```